eclair-core/src/main/scala/fr/acinq/eclair/payment/PaymentPacket.scala

```diff
@@ -241,9 +241,9 @@ object OutgoingPaymentPacket {
   sealed trait Upstream
   object Upstream {
     case class Local(id: UUID) extends Upstream
-    case class Trampoline(adds: Seq[UpdateAddHtlc]) extends Upstream {
-      val amountIn: MilliSatoshi = adds.map(_.amountMsat).sum
-      val expiryIn: CltvExpiry = adds.map(_.cltvExpiry).min
+    case class Trampoline(adds: Seq[(UpdateAddHtlc, TimestampMilli)]) extends Upstream {
```

Comment:
nit: it's probably worth introducing a `case class ReceivedHtlc(add: UpdateAddHtlc, receivedAt: TimestampMilli)`, that would be cleaner.